src/PhpDoc/PhpDocNodeResolver.php

```diff
@@ -170,6 +185,25 @@ private function resolveMethodTags(PhpDocNode $phpDocNode, NameScope $nameScope)
 		return $resolved;
 	}
 
+	/**
+	 * @param PhpDocNode $phpDocNode
+	 * @param NameScope $nameScope
+	 * @return array<string, \PHPStan\PhpDoc\Tag\TemplateTag>
+	 */
+	private function resolveTemplateTags(PhpDocNode $phpDocNode, NameScope $nameScope): array
+	{
+		$resolved = [];
+
+		foreach ($phpDocNode->getTemplateTagValues() as $tagValue) {
+			$resolved[$tagValue->name] = new TemplateTag(
+				$tagValue->name,
+				$this->typeNodeResolver->resolve($tagValue->bound, $nameScope)
```

Comment:
What does the `$tagValue->bound` contain if we have `@template T` without any bound in the phpDoc?

Reply:
It always contains a TypeNode. @JanTvrdik [suggested](https://github.com/phpstan/phpdoc-parser/pull/27#issuecomment-494804076) that it should be null in this case; I will make this change later.